src/action_intf.ml

```diff
@@ -21,23 +22,24 @@ module type Ast = sig
     | Progn          of t list
     | Echo           of string list
```

Comment:
Should the type of `Redirect` constructor also change?

Reply:
For redirect, it cannot change yet because `/dev/null` is allowed as a target in jbuild files. In dune files, this has to be written as `%{null}` so I think we need to wait for 2.0 before making this change.

For mkdir, I think external paths are allowed as long as they already exist. This is for use cases like `(mkdir %{env:foo})`.

Reply:
I see. Thanks. This might be worth a comment, esp. the redirect part.

Reply:
Done